src/dlgButtonSelect.cpp

```diff
@@ -0,0 +1,233 @@
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+
+#include "dlgButtonSelect.h"
+
+
+#include "Host.h"
+#include "HostManager.h"
+#include "mudlet.h"
+
+#include "pre_guard.h"
+#include <QFileDialog>
+#include <QInputDialog>
+#include <QMessageBox>
+#include <QPainter>
+#include <QtUiTools>
+#include <QGamepad>
+#include "post_guard.h"
+
+
+#define _DEBUG_
+
+dlgButtonSelect::dlgButtonSelect(QWidget * parent)
+: QDialog( parent )
+{
+    setupUi( this );
+
+    connect(buttonBox, SIGNAL(accepted()), this, SLOT(accept()));
+    connect(buttonBox, SIGNAL(rejected()), this, SLOT(reject()));
+
+    connect(QGamepadManager::instance(), &QGamepadManager::gamepadButtonPressEvent, this, slot_gamepadButtonPress);
+    connect(QGamepadManager::instance(), &QGamepadManager::gamepadButtonReleaseEvent, this, slot_gamepadButtonRelease);
+    connect(QGamepadManager::instance(), &QGamepadManager::gamepadAxisEvent, this, slot_gamepadAxisEvent);
+
+    connect(pressRadio, SIGNAL(pressed()), this, SLOT(slot_pressRadioPressed()));
+    connect(releaseRadio, SIGNAL(pressed()), this, SLOT(slot_releaseRadioPressed()));
+    connect(repickButton, SIGNAL(clicked()), this, SLOT(slot_repickButtonClicked()));
+
+    buttonSelected = false;
+}
+
+void dlgButtonSelect::slot_pressRadioPressed()
+{
+    pressThresholdSlider->setEnabled(true);
+}
+
+void dlgButtonSelect::slot_releaseRadioPressed()
+{
+    pressThresholdSlider->setDisabled(true);
+}
+
+void dlgButtonSelect::slot_repickButtonClicked()
+{
+    buttonSelected = false;
+    tabWidget->setCurrentIndex(0);
+    deviceEdit->setText("0");
+    buttonEdit->setText("0");
+    pressRadio->click();
+    pressThresholdSlider->setValue(75);
+    repickButton->setDisabled(true);
+}
+
+void dlgButtonSelect::slot_gamepadButtonPress(int deviceId, QGamepadManager::GamepadButton button, double value)
+{
+    lastEventLabel->setText(QString("Device %1: Button %2 was pressed to the %3 value").arg(deviceId).arg(button).arg(value));
+
+    if ( !buttonSelected ){
+        deviceEdit->setText(QString("%1").arg(deviceId));
+        tabWidget->setCurrentIndex(0);
+        buttonEdit->setText(QString("%1").arg(button));
+        buttonSelected = true;
+        repickButton->setEnabled(true);
+        return;
+    }
+
+    if ( button == 12 ){// dpad U
```

Comment:
:bulb: You might want to learn how to roll your own `QFlags` in the future - take a look at the Qt Documentation for that and you can see a simple example in the `mudlet` class header:

    enum StatusBarOption {
        statusBarHidden = 0x0,     // Currently not on display
        statusBarAutoShown = 0x1,  // Currently shown but to hide as soon as there is no text to display
        statusBarAlwaysShown = 0x2
    };
    Q_DECLARE_FLAGS(StatusBarOptions, StatusBarOption)

    StatusBarOptions             mStatusBarState;

and the later (outside the class definition):

    Q_DECLARE_OPERATORS_FOR_FLAGS(mudlet::StatusBarOptions)

Actually - after failing to find the details in my Qt 5.8 setup (it has `QGamePad` but not `QGamepadManager`) - I found from the on-line Qt **5.10** documentation:
> `enum QGamepadManager::GamepadButton`
> `flags QGamepadManager::GamepadButtons`
>
> The `GamepadButtons` type is a `typedef` for `QFlags<GamepadButton>`. It stores an OR combination of `GamepadButton` values.  

What that did not detail was all the enum values :frowning: